skema/skema_py/server.py

```diff
@@ -65,6 +71,46 @@ class System(BaseModel):
     )
 
 
+def system_to_enriched_system(system: System) -> System:
+    """Takes a System as input and enriches it with comments by running the Rust comment extractor."""
+
+    # The skema_rs_path is the path to the Rust comment extractor.
+    # It is used to set the cwd when running subprocess to simplify the execution process.
+    skema_rs_path = Path(__file__).parent.parent / "skema-rs" / "comment_extraction"
+
+    # Check if Rust is installed on the system. If not, then return the system as is.
+    if not which("cargo"):
+        return system
+
+    # Create a temporary directory to store the input source files.
+    # The Rust comment_extractor requires that the input exists on disk.
+    tmp = tempfile.TemporaryDirectory()
+    tmp_path = Path(tmp.name)
+    
+    comments = {"files": {}}
+    for file, blob in zip(system.files, system.blobs):
+        file_path = Path(tmp_path, system.root_name or "", file)
+        file_path.parent.mkdir(parents=True, exist_ok=True)
+        file_path.write_text(blob)
+
+        result = subprocess.run(
+            ["cargo", "run", "--", str(file_path)],
```

Comment:
1. Does this include an invocation of the Rust runnable?

Reply:
Yes, the full invocation would look like
`cargo run -- /tmp/root_name/example.py`.
**cargo run** is the Rust runnable, **--** specifies that the code to run is in the cwd, and  **/tmp/root_name/example.py** is the path to the source file.